public/scripts/extensions/regex/engine.js

```diff
@@ -7,6 +7,54 @@ export {
     runRegexScript,
 };
 
+/**
+ * @enum {number} Regex scripts types
+ */
+export const SCRIPT_TYPES = {
+    GLOBAL: 0,
+    SCOPED: 1,
+};
+
+/**
+ * @typedef {import('../../char-data.js').RegexScriptData} RegexScript
+ */
+
+/**
+ * @typedef {object} GetRegexScriptsOptions
+ * @property {boolean} allowedOnly only return allowed scripts
+ */
+
+/**
+ * Retrieves the list of regex scripts by combining the scripts from the extension settings and the character data
+ *
+ * @param {GetRegexScriptsOptions} option
+ * @returns {RegexScript[]} An array of regex scripts, where each script is an object containing the necessary information.
+ */
+export function getRegexScripts(option = { allowedOnly: false }) {
+    return [...(getScriptsByType(SCRIPT_TYPES.GLOBAL, option)), ...(getScriptsByType(SCRIPT_TYPES.SCOPED, option))];
+}
```

Comment:
This (and the other places below) are intended to have the same options to be able to call into, right?
Because you added the jsdoc `GetRegexScriptsOptions`.
If we ever add more options, you'd need to set the new defaults in all functions too then.
I'd say if you already go the way of `options` object instead of manually listing them, define a global freezed `DEFAULT_REGEX_SCRIPTS_OPTIONS` or something.

Question also is, if you already have a `getScriptsByType`, this could be simpler by calling that for each `SCRIPT_TYPES` enum value and flatten it.